shared/utils.js

```diff
@@ -0,0 +1,204 @@
+import fetch from 'node-fetch'
+import {createClient} from '@commercetools/sdk-client'
+import {createRequestBuilder} from '@commercetools/api-request-builder'
+import {createHttpMiddleware} from '@commercetools/sdk-middleware-http'
+import {createQueueMiddleware} from '@commercetools/sdk-middleware-queue'
+import {createAuthMiddlewareForClientCredentialsFlow} from '@commercetools/sdk-middleware-auth'
+import _ from 'lodash'
+import util from 'util'
+
+function createCtpClient({clientId, clientSecret, projectKey, concurrency = 20}) {
+    const AUTH_HOST = process.env.AUTH_HOST || 'https://auth.europe-west1.gcp.commercetools.com'
+    const API_HOST = process.env.API_HOST || 'https://api.europe-west1.gcp.commercetools.com'
+    const authMiddleware = createAuthMiddlewareForClientCredentialsFlow({
+        host: AUTH_HOST,
+        projectKey,
+        credentials: {
+            clientId,
+            clientSecret
+        },
+        fetch
+    })
+
+    const httpMiddleware = createHttpMiddleware({
+        maskSensitiveHeaderData: true,
+        host: API_HOST,
+        enableRetry: true,
+        fetch
+    })
+
+    const queueMiddleware = createQueueMiddleware({
+        concurrency
+    })
+
+    return createClient({
+        middlewares: [
+            authMiddleware,
+            httpMiddleware,
+            queueMiddleware
+        ]
+    })
+}
+
+export function setUpClient(config) {
+    const ctpClient = createCtpClient(config.ctp)
+    const customMethods = {
+        get builder() {
+            return getRequestBuilder(config.ctp.projectKey)
+        },
+
+        buildRequestOptions(uri, method = 'GET', body = undefined) {
+            return {
+                uri,
+                method,
+                body,
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                }
+            }
+        },
+
+        async * fetchPages ({ variables, endpoint }) {
+            const originalWhere = variables.where
```

Comment:
just wondering if it's a good practice to put withTotal=false for the pagination requests.